internal/build/phase_config_provider_test.go

```diff
@@ -32,7 +33,7 @@ func testPhaseConfigProvider(t *testing.T, when spec.G, it spec.S) {
 			expectedBuilderName := "some-builder-name"
 			fakeBuilder, err := fakes.NewFakeBuilder(fakes.WithName(expectedBuilderName))
 			h.AssertNil(t, err)
-			lifecycle := fakeLifecycle(t, false, fakes.WithBuilder(fakeBuilder))
+			lifecycle := newLifecycle(t, false, fakes.WithBuilder(fakeBuilder))
```

Comment:
Changed the function name since the lifecycle isn't really a fake - it is an actual `*build.Lifecycle`, it just has a fake builder.

Reply:
I agree that `fakeLifecycle` is really confusing but I'm not sure that `newLifecycle` is clear enough because it doesn't indicate that the builder is fake.
In this call, it's clear since we pass `fakes.WithBuilder(fakeBuilder)` but in most cases, we just pass the first two arguments to the function and then it's not clear that we're using a fake builder.
Unfortunately, I can't think of a better name (`newLifecycleWithFakeBuilder` is probably too long).

Reply:
Maybe `testLifecycle`?

Reply:
Going with `newTestLifecycle`...